### PR TITLE
[release/2.1] Servicing SignedCms for Signed NuGet Packages

### PR DESCRIPTION
**Pull request by bartonjs**
Backports of #32666 (registration for alternate RSA signature identifiers) and #32704 (indefinite-length encoded content) to release/2.1

#### Description
Windows CMS (and therefore .NET Framework) is more permissive on SignedCms content than the .NET Core implementation of SignedCms.

OIDs: The .NET Foundation has created Signed NuGet packages which validate on .NET Framework, but not .NET Core, as .NET Core does not recognize the signature algorithm identifier.

Encoding: The .NET Foundation has created Signed NuGet packages which cannot be countersigned by .NET Core.

#### Customer Impact
Customers will be unable to work with RSA-signed packages signed by the .NET Foundation using the X.509 signature algorithm identifier instead of the CMS algorithm identifier.

Packages signed by The .NET Foundation may decode correctly, but cannot be re-encoded.

#### Regression?
The OID entries are a behavioral regression from .NET Framework, but not a regression within .NET Core.

The indefinite-length encoding content issue was a regression from #30432 (which released in 2.1.5).

#### Packaging reviewed? 
Packaging considered, reviewers tagged.

#### Risk
Low.  Existing unit tests provide assurance that the formats written by .NET Core and .NET Framework still behave as expected, new unit tests guard against regressions from variants encountered from the signing service set up by The .NET Foundation based on Bouncy Castle.

---

Since both of these changes require the package harvesting manifest to be updated, the package index to be updated, the version of System.Security.Cryptography.Pkcs to be bumped, and System.Security.Cryptography.Pkcs to be registered for build/publish, they have been bundled into one change.  They could be separated if desired.

### CODE REVIEW

File: external/harvestPackages/harvestPackages.props [hunk 1]
```diff
@@ -116,7 +116,7 @@
   <Version>4.5.0</Version>
 </PackageReference>
 <PackageReference Include="System.Security.Principal.Windows">
-  <Version>4.5.0</Version>
+  <Version>4.5.1</Version>
```
Comment (safern) [inline external/harvestPackages/harvestPackages.props]: I guess you just ran the updateHarvest target?

Comment (bartonjs) [reply external/harvestPackages/harvestPackages.props]: Yep

**Comment by bartonjs** [comment]
@dotnet-bot Test Windows x64 Debug Build please (network timeout)
Test OSX x64 Debug Build please (#28360)

**Comment by vivmishra** [comment]
Approved for 2.1.7 and 2.2.1

**Pull request merged**